app/src/main/java/io/lbry/browser/MainActivity.java

```diff
@@ -2781,15 +2781,12 @@ private void showSignedInUser() {
     }
 
     private Fragment getCurrentFragment() {
-        int backCount = getSupportFragmentManager().getBackStackEntryCount();
-        if (backCount > 0) {
-            try {
-                return getSupportFragmentManager().getFragments().get(backCount - 1);
-            } catch (IndexOutOfBoundsException ex) {
-                return null;
-            }
-        }
-        return null;
+        List<Fragment> fragmentList = getSupportFragmentManager().getFragments();
+
+        if (fragmentList.size() > 0)
+            return fragmentList.get(fragmentList.size() - 1);
+        else
+            return null;
```

Comment:
I used the back stack entry count here because not all the fragments we display are added to the back stack. If you're 100% sure this will still work regardless, then this is ok. Everything else looks good.

Reply:
That part of the code was giving me problems because after the first time I clicked on a notification and wanted to view some other content, there was a silent IndexOutOfBoundsException. I think that silent exception -I mean, catched, but nothing else- could be related to notification items not working when clicked.

I tested this new code for this bug, but not in other conditions. I will be debugging it for longer and commit and request another review if needed.

Reply:
Latest commit 96606f5 simplifies getting the fragment which is currently on main activity by querying the OS for it. Main activity is using a frame layout as a parent layout where fragments can be added. That allows app to simply request which fragment is there, instead of travelling the backstack or the list of fragments.